File: client/src/app/models/cloud/cloud-file.ts
import { FileType } from "src/app/shared/enumerations/file.enum";
import { BaseModel } from "../base/base-model";

export class CloudFile extends BaseModel {
    public fileName = "";
    public originalFileName = "";
    public fileExtension = "";
    public size = 0;
    public directoryId = "";
    public url = "";
}

export class CloudFileUI extends CloudFile {
    public selected = false;
    public fileType = FileType.Other;
    public isDeleting = false;
    public canChangeSelected = true;
    public underCopyOrCut = false;
}

File: server/Src/Services/Open/OpenVN.Domain/Entities/Cloud/CloudFile.cs
namespace OpenVN.Domain
{
    [Table("cloud_file")]
    public class CloudFile : PersonalizedEntity
    {
        public string FileName { get; set; }

        public string OriginalFileName { get; set; }

        public string FileExtension { get; set; }

        public long Size { get; set; }

        public long DirectoryId { get; set; }

        public CloudFile()
        {
        }

        public CloudFile(string fileName, string originalFileName, string fileExtension, long size, long directoryId)
        {
            FileName = fileName;
            OriginalFileName = originalFileName;
            FileExtension = fileExtension;
            Size = size;
            DirectoryId = directoryId;
        }

        public CloudFile(string fileName, string originalFileName, string fileExtension, long size, long directoryId, DomainEvent @event)
        {
            FileName = fileName;
            OriginalFileName = originalFileName;
            FileExtension = fileExtension;
            Size = size;
            DirectoryId = directoryId;
            AddDomainEvent(@event);
        }
    }
}
